### PR TITLE
Added Slutmachine

### PR DESCRIPTION
**Pull request by exp111**
Should work

### CODE REVIEW

File: TS3GameBot/CommandStuff/CommandBase.cs [hunk 1]
```diff
@@ -22,6 +22,8 @@ public abstract class CommandBase
 
 		public bool Restricted { get; set; } = false;
 
+		public bool NeedsRegister { get; set; } = false;
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/CommandBase.cs]: Maybe change the default to true, depending on how many commands require you to be registered!

File: TS3GameBot/CommandStuff/CommandBase.cs [hunk 2]
```diff
@@ -22,6 +22,8 @@ public abstract class CommandBase
 
 		public bool Restricted { get; set; } = false;
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/CommandBase.cs]: Defaults to false, assigning false is not required

File: TS3GameBot/CommandStuff/Commands/CommandWallet.cs [hunk 3]
```diff
@@ -12,21 +12,17 @@ class CommandWallet : CommandBase
...
-			if(tempPlayer == null)
-			{
-				CommandManager.AnswerCall(message, Responses.NotRegistered);
-				return false;
-			}
+			CasinoPlayer player = DbInterface.GetPlayer(message.InvokerUid);
 
 			outMessage.Append("\n").
-				Append(CommandManager.ClientUrl(tempPlayer.Id, tempPlayer.Name) + ": You have " + DbInterface.GetPlayer(message.InvokerUid).Points + " in your Wallet!");
+				Append(CommandManager.ClientUrl(player.Id, player.Name) + ": You have " + player.Points + " in your Wallet!");
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandWallet.cs]: Use string interpolation

File: TS3GameBot/CommandStuff/Commands/CommandDaily.cs [hunk 4]
```diff
@@ -23,43 +24,34 @@ public override bool Execute(List<string> args, TextMessage message)
 
 			outMessage.Append("\n");
 
-			if (myPlayer != null)
+			//myPlayer.LastDaily <= DateTime.Now.AddDays(-1)
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandDaily.cs]: Is this comment really necessary?

File: TS3GameBot/CommandStuff/CommandManager.cs [hunk 5]
```diff
@@ -25,6 +26,7 @@ public static void RegisterCommands()
 			RegisterCommand(new CommandDaily("daily", "Get your frickinnn Daily reeeeeword"));
 			RegisterCommand(new CommandWallet("wallet", "Get the amount of cash you have"));
 			RegisterCommand(new CommandTransfer("transfer", "Transfer your money 2 peepz"));
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/CommandManager.cs]: Why don't you put the label and description directly into the command classes?

File: TS3GameBot/CommandStuff/Commands/CommandSpin.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+		public int Cost { get; private set; }
+		public int MagicNumber { get; private set; }
+
+
+		public CommandSpin(string label, string description) : base(label, description)
+		{
+			this.Usage = "";
+			this.NeedsRegister = true;
+
+			this.Cost = 25;
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandSpin.cs]: You could make those proerties constant if they are only assigned here.

File: TS3GameBot/CommandStuff/Commands/CommandSpin.cs [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+				#############
+				# 5 | 6 | 5 #
+				#############
+
+				And they won a meh Price...
+			  */
+
+			if (myPlayer.Points < this.Cost) //Not enough points
+			{
+				CommandManager.AnswerCall(message, Utils.Utils.ApplyColor(Color.Red) + "\nNot enough Points in your Wallet![S](get fucked)[/S][/COLOR]");
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandSpin.cs]: I don't know about the message...

File: TS3GameBot/CommandStuff/Commands/CommandSpin.cs [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+			//Generate Numbers
+			Random run = new Random();
+			int firstNum = run.Next(1, 9);
+			int secondNum = run.Next(1, 9);
+			int thirdNum = run.Next(1, 9);
+
+			int price = 0;
+
+			//Create Message
+			outMessage.Append(CommandManager.ClientUrl(myPlayer.Id, myPlayer.Name) + " has paid " + this.Cost + " to spin the Slot Machine!");
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandSpin.cs]: Use string interpolation.

File: TS3GameBot/CommandStuff/Commands/CommandSpin.cs [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+			outMessage.Append("#############\n");
+			outMessage.Append("# " + firstNum + " | " + secondNum + " | " + thirdNum + " #\n");
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandSpin.cs]: Use string interpolation.

File: TS3GameBot/CommandStuff/Commands/CommandSpin.cs [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+			}
+			else //Nothing
+			{
+				outMessage.Append("\nYou've won [S]a price[/S] [B]nothing[/B]!\n");
+				price = 0;
+			}
+
+			if (!DbInterface.AlterPoints(myPlayer, price - this.Cost)) //Give/Take money
+			{ //Can't change points for some reason
+				CommandManager.AnswerCall(message, "Shitty mcshitfuck");
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandSpin.cs]: You should give a more informative response.

File: TS3GameBot/CommandStuff/Commands/CommandSpin.cs [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+				throw new Exception("Shitty mcschit fuck again");
+			}
+			DbInterface.SaveChanges();
+			outMessage.Append("You now have " + myPlayer.Points + " Points!");
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandSpin.cs]: Use string interpolation.

File: TS3GameBot/CommandStuff/Commands/CommandDaily.cs [hunk 12]
```diff
@@ -23,43 +24,34 @@ public override bool Execute(List<string> args, TextMessage message)
...
-					CommandManager.AnswerCall(message, outMessage.ToString());
-					return false;
-				}
-				CommandManager.AnswerCall(message, outMessage.ToString());
-				return true;
 			}
 			else
 			{
-				CommandManager.AnswerCall(message, "\nYou are not registered yet!\nUse " + CommandManager.CmdIndicator + CommandManager.Commands["register"].Label + " to register yourshelf.");
+				outMessage.Append(CommandManager.ClientUrl(myPlayer.Id, myPlayer.Name) + ": You already received your Daily reward " + Utils.Utils.MsToM((DateTime.Now - myPlayer.LastDaily).TotalMilliseconds) + " ago!\nCome back in " + Utils.Utils.MsToM((DateTime.Now.AddDays(1).Date - DateTime.Now).TotalMilliseconds) + ".");
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandDaily.cs]: Use string interpolation.

File: TS3GameBot/CommandStuff/Commands/CommandDaily.cs [hunk 13]
```diff
@@ -23,43 +24,34 @@ public override bool Execute(List<string> args, TextMessage message)
...
-						case Error.SAVEERROR:
-							outMessage.Append("Changes could not be saved to the Database!");
-							break;
-						case Error.UNKNOWN:
-						default:
-							outMessage.Append("An Unknown Error Occured");
-							break;
-					}
+					case Error.OK:
+						outMessage.Append(this.DailyReward + " Points have been added to your Wallet!\nYou now have " + myPlayer.Points + " Points in your Wallet!");
```
Comment (Cavadeca) [inline TS3GameBot/CommandStuff/Commands/CommandDaily.cs]: Use string interpolation.

File: TS3GameBot/CommandStuff/CommandManager.cs (hunk 5 above)
Comment (MrDj200) [reply TS3GameBot/CommandStuff/CommandManager.cs]: Cuz u showed this like dis

**Comment by exp111** [comment]
let's just not use his suggestions k?

File: TS3GameBot/CommandStuff/CommandManager.cs (hunk 5 above)
Comment (Cavadeca) [reply TS3GameBot/CommandStuff/CommandManager.cs]: What?

**Comment by MrDj200** [comment]
de82aeaa32c6c231501627a0b3589cdebb6f8112 Did the things manually => **CLOSED**

**Pull request closed without merging**